xacc/service/xacc_service.cpp

```diff
@@ -60,6 +60,7 @@ void ServiceAPI_Finalize() {
     if (serviceAPIInitialized) {
         serviceAPIInitialized = false;
         serviceRegistry->finalize();
+        serviceRegistry.release();
```

Comment:
`release` gives up the pointer for someone else to hold on to (in this case, it's leaking memory). I think you want `reset` or `= {}` or `= nullptr`.

Reply:
Thanks, will fix.

I also will look into why the tests are defaulting. I wonder if something is accessing the service registry after finalize()